File: InMatlabPath/loadSerializedData.m
function [next_data_start, data] = loadSerializedData(mapped_file, data_start)
    data_end = data_start + 4 - 1;
    data_length = typecast(mapped_file.Data(data_start:data_end),'uint32');
    % disp("Custom data length: " + string(data_length));
    data_start = data_end + 1; data_end = data_start + data_length - 1;
    serialized_data = mapped_file.Data(data_start:data_end);
    data = hlp_deserialize(serialized_data);
    next_data_start = data_end + 1;
end

File: report/py_Mouse2AFC/common/loadSerializedData.py
import numpy as np
from .hlp_deserialize import hlp_deserialize

def loadSerializedData(mapped_file, data_start):
    data_end = data_start + 4
    data_length = np.frombuffer(mapped_file[data_start:data_end],
                                dtype=np.uint32)[0]
    # print("Custom data length: " + str(data_length))
    data_start = data_end
    data_end = data_start + data_length
    # print("Start:", data_start, " - End:", data_end)
    serialized_data = mapped_file[data_start:data_end]
    data = hlp_deserialize(serialized_data)
    next_data_start = data_end
    return next_data_start, data

